src/shared/cloudformation/templateRegistry.ts

```diff
@@ -0,0 +1,143 @@
+/*!
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import * as path from 'path'
+import { fileURLToPath, pathToFileURL } from 'url'
+import * as vscode from 'vscode'
+import { FileWatcherListener } from '../utilities/fileSystemWatcher'
+import { CloudFormation } from './cloudformation'
+
+let templateRegistry: CloudFormationTemplateRegistry | undefined
+
+export function getTemplateRegistry(): CloudFormationTemplateRegistry {
```

Comment:
Is there a way we can avoid global functions like this? What is the lifecycle of registry activation? Is there threading concerns? What if activation happens from multiple threads?

Reply:
This is our current convention. Activation should be single-threaded and kicked off in the [extension.ts](https://github.com/aws/aws-toolkit-vscode/blob/master/src/extension.ts#L59) file.